### ISSUE TITLE
`polars collect` breakes further operations

### ISSUE THREAD

**Issue by maxim-uvarov** [nushell/nushell#12828]
### Describe the bug

@ayax79 I found another bug? (Execute in the order as given below.) Without `$a | polars collect`, the last command works as expected.

Though, I don't fully understand if `polars collect` is needed now. But anyway, it would be better if you personally took a look into it; maybe it is important.

### How to reproduce

```
> let a = [[a b]; [1 a] [2 b] [3 c] [4 d]] | polars into-df

> $a | polars collect
╭─#─┬─a─┬─b─╮
│ 0 │ 1 │ a │
│ 1 │ 2 │ b │
│ 2 │ 3 │ c │
│ 3 │ 4 │ d │
╰─#─┴─a─┴─b─╯

> $a
 | polars reverse
 | polars with-column [
      ((polars col a) * 2 | polars as double_a)
      ((polars col a) / 2 | polars as half_a)
 ]
 | polars collect
Error:   × Dataframe 13a4ee8a-5bea-41f2-9e1c-67dda5c789b9 not found in cache
```

### Expected behavior

polars to work

### Configuration

| key                | value                                                      |
| ------------------ | ---------------------------------------------------------- |
| version            | 0.93.1                                                     |
| major              | 0                                                          |
| minor              | 93                                                         |
| patch              | 1                                                          |
| branch             | main                                                       |
| commit_hash        | c54d223ea0b14f180b1793fa539419854c2e014f                   |
| build_os           | macos-aarch64                                              |
| build_target       | aarch64-apple-darwin                                       |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09)                        |
| rust_channel       | 1.77.2-aarch64-apple-darwin                                |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26)                        |
| build_time         | 2024-05-07 05:50:32 +00:00                                 |
| build_rust_channel | release                                                    |
| allocator          | mimalloc                                                   |
| features           | dataframe, default, sqlite, system-clipboard, trash, which |
| installed_plugins  | explore, image, inc, polars                                |

**Comment by maxim-uvarov** [comment]
I think I understood what the problem was: `polars collect` errored on eager dataframes with not very informative message. 

```
> $a
 | polars reverse
 | polars with-column [
      ((polars col a) * 2 | polars as double_a)
      ((polars col a) / 2 | polars as half_a)
 ]
 | polars collect
Error:   × Dataframe 13a4ee8a-5bea-41f2-9e1c-67dda5c789b9 not found in cache
```

`polars with-column` or `polars reverse` already collected the lazyframe before.

**Comment by haras-unicorn** [comment]
same thing with `polars into-nu`
im personally using it to batch requests to a server from a large csv file

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.95.0                                                            |
| major              | 0                                                                 |
| minor              | 95                                                                |
| patch              | 0                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.78.0 (9b00956e5 2024-04-29) (built from a source tarball) |
| cargo_version      | cargo 1.78.0                                                      |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, system-clipboard, trash                          |
| installed_plugins  | polars 0.95.0                                                     |

**Comment by haras-unicorn** [comment]
did a bit more research - it seems the dataframe stays in cache after `polars into-nu` but it dissapears right after i try doing anything with the table and lowering my batch size seems to fix it??
its as if the longer you wait the higher the chance of the dataframe staying in cache?
idk but its really weird behaviour imo

**Comment by fdncred** [comment]
We're hoping [this PR](https://github.com/nushell/nushell/pull/13286) fixed the issue of the cache disappearing

**Comment by maxim-uvarov** [comment]
> We're hoping https://github.com/nushell/nushell/pull/13286 fixed the issue of the cache disappearing

Nope, it didn't fix. 

@ayax79 would you please take a look at this? 

I'm on:
```
> version | select version installed_plugins
╭───────────────────┬──────────────────────╮
│ version           │ 0.96.1               │
│ installed_plugins │ polars 0.96.1, regex │
╰───────────────────┴──────────────────────╯
```

and this example errors:

```
> let a = [[a b]; [1 a] [2 b] [3 c] [4 d]] | polars into-df
> $a | polars collect
> $a
Error:   × Dataframe e8a51f5c-5865-4dc2-af4a-b30a0dbbae3f not found in cache
```

**Comment by fdncred** [comment]
@maxim-uvarov could you check again. I just tested this on the latest main branch with the latest polars plugin.
```nushell
❯ let a = [[a b]; [1 a] [2 b] [3 c] [4 d]] | polars into-df
❯ $a
╭─#─┬─a─┬─b─╮
│ 0 │ 1 │ a │
│ 1 │ 2 │ b │
│ 2 │ 3 │ c │
│ 3 │ 4 │ d │
╰───┴───┴───╯
❯ $a | polars collect
╭─#─┬─a─┬─b─╮
│ 0 │ 1 │ a │
│ 1 │ 2 │ b │
│ 2 │ 3 │ c │
│ 3 │ 4 │ d │
╰───┴───┴───╯
```

**Comment by maxim-uvarov** [comment]
@fdncred thank you for checking!

`polars collect` breaks the initial variable, and in your example you used it at the last step without checking it effects. 

I compiled the latest nushell and polars from the main, and confirm that I still experience the problem.

<img width="785" alt="image" src="https://github.com/user-attachments/assets/b617eb0f-411d-473a-9ab4-d39e7d9620f2">

**Comment by fdncred** [comment]
oh, you're saying _after_ you use `polars collect` the variable isn't in the cache. I missed that. Paging @ayax79 for insight.

**Comment by ayax79** [comment]
I think the plugin framework is incorrectly calling drop on values that have been saved as a variable. I think I'll need to reach out to @devyn on advice on this one. 

<img width="1459" alt="Screenshot 2024-08-14 at 09 47 24" src="https://github.com/user-attachments/assets/f8c02f9f-62e0-4037-aac1-cce196d8afb4">

**Comment by ayax79** [comment]
Figured it out with @devyn's help. When calling collect on an eager frame, the eager frame is just returned. I needed to make sure the the plugin cache reference count was incremented first.

Fix: #13624
